src/Nancy/DefaultObjectSerializer.cs

```diff
@@ -46,13 +78,15 @@ public object Deserialize(string sourceString)
             try
             {
                 var inputBytes = Convert.FromBase64String(sourceString);
-
-                var formatter = new BinaryFormatter();
-
-                using (var inputStream = new MemoryStream(inputBytes, false))
+                var json = Encoding.UTF8.GetString(inputBytes);
+                //When deserializing objects it will have a TypeObject property
+                if (json.Contains("TypeObject"))

```

Comment:
Move this to a helper method with a descriptive name, something like "ContainsTypeDecription(json)` or similar